microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api.md

```diff
@@ -0,0 +1,213 @@
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: pwa
+ms.date: 11/25/2024
+---
+# Provide in-app purchases with the Digital Goods API
+
+If your Progressive Web App (PWA) is listed in the Microsoft Store, you can provide in-app products and subscriptions by using the Digital Goods API and the Payment Request API.
+
+
+<!-- ====================================================================== -->
+## Digital Goods API
+
+The Digital Goods API is an interface between your PWA app and Microsoft Store.  The Digital Goods API supports:
+* Querying the details of a digital item from the Microsoft Store backend, such as the item's name, description, and regional price.
+* Consuming or acknowledging purchases.
+* Checking the digital items that are currently owned by the user.
+* Checking the purchase history of the user.
+
+See:
+* [Digital Goods API For Microsoft Store PWA Explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/PwaDigitalGoods/explainer.md)
+* [Digital Goods API: Draft Community Group Report](https://wicg.github.io/digital-goods/)
+
+
+<!-- ====================================================================== -->
+## Payment Request API
+
+The Payment Request API⁠⁠ handles the actual payment transaction when a purchase is made by user.  The Payment Request API uses the item details that the Digital Goods API provides, to make the in-app purchase by using whichever billing payment method the user has set up at the Microsoft Store.
+
+See [Payment Request API](https://www.w3.org/TR/payment-request/) at W3C.
+
+
+<!-- ====================================================================== -->
+## Checking whether the Digital Goods API is available
+
+To detect whether you've correctly enabled the API on your website, check for the `getDigitalGoodsService` method in the window object:
+
+```javascript
+if ('getDigitalGoodsService' in window) {
+  // Digital Goods API is supported!
+} else {
+  console.log('DigitalGoodsService is not available.');
+  // Use other payment method
+}
+```
+
+
+<!-- ====================================================================== -->
+## Connecting to the Microsoft Store Billing service (`getDigitalGoodsService` method)
+
+Use the `getDigitalGoodsService` method to connect to the Microsoft Store Billing service.
+
+The Digital Goods API was designed to be compatible with various browsers and digital stores, similar to how the Payment Request API is browser-agnostic and can be used with different payment providers.  To retrieve an instance of the service for Microsoft Store Billing, pass the string `"https://store.microsoft.com/billing"` as the payment method to the `getDigitalGoodsService` method.
+
+If the method throws an error, the Microsoft Store Billing payment method is not available (such as when the user is accessing your PWA through the browser).  Alternatively, consider providing a different payment method for transactions.
```

Comment:
I think we should expand on the reasons that might make the payment method not available. When is the method available exactly? What are the requirements for it to be available? PWA installed from the store? Any other required settings?

Reply:
doc'd:
https://review.learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/digital-goods-api?branch=pr-en-us-3324#connecting-to-the-microsoft-store-billing-service-windowgetdigitalgoodsservice-method
@captainbrosset